conans/server/rest/api_v2.py

```diff
@@ -0,0 +1,33 @@
+from bottle import Bottle
+
+from conans.server.rest.api_v1 import ApiV1
+from conans.server.rest.controllers.delete_controller import DeleteController
+from conans.server.rest.controllers.ping_controller import PingController
+from conans.server.rest.controllers.search_controller import SearchController
+from conans.server.rest.controllers.users_controller import UsersController
+from conans.server.rest.controllers.v2.conan_controller import ConanControllerV2
+
+
+class ApiV2(ApiV1):
+
+    def __init__(self, credentials_manager, server_version, min_client_compatible_version,
+                 server_capabilities):
+
+        self.credentials_manager = credentials_manager
+        self.server_version = server_version
+        self.min_client_compatible_version = min_client_compatible_version
+        self.server_capabilities = server_capabilities
+        Bottle.__init__(self)
+
+    def setup(self):
+        self.install_plugins()
+
+        # Capabilities in a ping
+        PingController("").attach_to(self)
+
+        SearchController("/conans").attach_to(self)
```

Comment:
I am a bit surprised that SearchController and DeleteController are not affected at all by package revisions. Is it not possible to search binaries of a given revisions, or to delete a recipe+binaries of a revision?

Reply:
Good question. The interface of the controllers is almost the same, I've modified them to add a new field "revision" (optional). It builds a reference object with the revision/s and then everything works exactly the same because of the storage object resolving different paths. 

But I just spotted a bug and probably we want to remove the "remove_packages_files" from the server too.